### ISSUE TITLE
`tf.raw_ops.StatelessMultinomial` produces inconsistent results on CPU/GPU

### ISSUE THREAD

**Issue by creakseek** [tensorflow/tensorflow#58450]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Source

binary

### Tensorflow Version

2.9.2

### Custom Code

Yes

### OS Platform and Distribution

Colab

### CUDA/cuDNN version

11.2.152, 8.1.1

### Current Behaviour?

Stateless sampling produced different results on CPU/GPU with the same seed. 
Related: #58445 

### Standalone code to reproduce the issue

```python
with tf.device("/cpu:0"):
  samples = tf.raw_ops.StatelessMultinomial(logits=tf.math.log([[0.5, 0.5]]), num_samples=10, seed=(7, 17))
  print(samples)

with tf.device("/gpu:0"):
  samples = tf.raw_ops.StatelessMultinomial(logits=tf.math.log([[0.5, 0.5]]), num_samples=10, seed=(7, 17))
  print(samples)
```

### Relevant log output

```shell
tf.Tensor([[1 0 1 0 0 0 1 1 0 0]], shape=(1, 10), dtype=int64)
tf.Tensor([[0 0 0 1 0 1 0 1 1 1]], shape=(1, 10), dtype=int64)
```
</details>

**Comment by creakseek** [comment]
In comparison, `tf.raw_ops.StatelessParameterizedTruncatedNormal` works well.

**Comment by bhack** [comment]
@wangpengmit   @cantonios  Same as https://github.com/tensorflow/tensorflow/issues/58445 it is reproducible only in eager mode.

```python
import tensorflow as tf
tf.config.set_soft_device_placement(False)


with tf.device("/cpu:0"):
  samples = tf.raw_ops.StatelessMultinomial(logits=tf.math.log([[0.5, 0.5]]), num_samples=10, seed=(7, 17))
  print(samples)

with tf.device("/gpu:0"):
  samples = tf.raw_ops.StatelessMultinomial(logits=tf.math.log([[0.5, 0.5]]), num_samples=10, seed=(7, 17))
  print(samples)

tf.compat.v1.disable_eager_execution()
sess = tf.compat.v1.Session()

with tf.device("/cpu:0"):
  x = tf.raw_ops.StatelessMultinomial(logits=tf.math.log([[0.5, 0.5]]), num_samples=10, seed=(7, 17))
  y = sess.run(x)
  print(y)

with tf.device("/gpu:0"):
  x = tf.raw_ops.StatelessMultinomial(logits=tf.math.log([[0.5, 0.5]]), num_samples=10, seed=(7, 17))
  y = sess.run(x)
  print(y)
```

```python
tf.Tensor([[1 0 1 0 0 0 1 1 0 0]], shape=(1, 10), dtype=int64)
tf.Tensor([[0 0 0 1 0 1 0 1 1 1]], shape=(1, 10), dtype=int64)
[[1 0 1 0 0 0 1 1 0 0]]
[[1 0 1 0 0 0 1 1 0 0]]
```

**Comment by mohantym** [comment]
Hi @creakseek !
I was not getting any issue in graph mode as said in above comment.
But in Eager mode. It is replicating .  
Attached [gist](https://colab.sandbox.google.com/gist/mohantym/a511dedd9b8bc7b1f77eb55d63be9d90/git_58450.ipynb#scrollTo=AT8miYKr2eqQ) for reference.

Hi @SuryanarayanaY !
Could you look at this issue. 
Attached gist in [2.8](https://colab.sandbox.google.com/gist/mohantym/614816f0d062d0c7d9e44c9df417292d/git_58450.ipynb#scrollTo=3G5sXL-r2Zdc), [2.9](https://colab.sandbox.google.com/gist/mohantym/2615044fb8b75c5d66ece172d245b64b/git_58450.ipynb#scrollTo=dQN2JTi12_aq) ,[2.10](https://colab.sandbox.google.com/gist/mohantym/f90518300e9d43bb15d122dce1efcb5c/git_58450.ipynb#scrollTo=dQN2JTi12_aq) and  [nightly](https://colab.sandbox.google.com/gist/mohantym/f90518300e9d43bb15d122dce1efcb5c/git_58450.ipynb#scrollTo=dQN2JTi12_aq) for reference.

Thank you!

**Comment by wangpengmit** [comment]
Marked as duplicating #58445 .
